src/Helpers/LocalMachineHelper.php

```diff
@@ -124,14 +122,18 @@ private function configureProcess(Process $process, string $cwd = NULL, ?bool $p
     return $process;
   }
 
-  private function executeProcess(Process $process, callable $callback = NULL, ?bool $printOutput = TRUE): Process {
-    if ($callback === NULL && $printOutput !== FALSE) {
-      $callback = function (mixed $type, mixed $buffer): void {
+  private function executeProcess(Process $process, callable $callback = NULL, bool $printOutput = TRUE): Process {
+    if ($callback === NULL && $printOutput) {
+      $callback = function (string $type, iterable|string $buffer): void {
         $this->output->write($buffer);
       };
     }
-    $process->start();
-    $process->wait($callback);
+    if ($process->getInput()) {
+      $this->runAsync($process);
+    }
+    else {
+      $process->run($callback);
```

Comment:
I need to examine the code path for push:artifact and similar commands. If it's runAsync, I'm not surprised it's broken since we're missing the callback. But even adding the callback didn't fix it.

It looks like starting with the callback should be the same as waiting with it. I'm not sure why we did it the way we did initially.

Reply:
Ah, it actually appears that the Symfony 6 upgrade caused the side effects. This code might work fine, though we probably need to pass the callback to run(), and generally make this more consistent.